games-server/crossfire-server/files/crossfire-server-1.75.0-incompatible-func-pointers.patch

```diff
@@ -0,0 +1,175 @@
+Struct wanted (void)() function pointers. GCC-15 says no.
+We compromised on (void)(const char*) and giving no argument to functions
+that take no argument. Hopefully, when someone upstream adds
+a function with num_args > 1, it would be a problem of upstream
+https://bugs.gentoo.org/949574
+--- a/include/libproto.h
++++ b/include/libproto.h
+@@ -64,7 +64,7 @@
+ extern int64_t new_exp(const object *ob);
+ extern int has_ability(const object *ob);
+ extern void init_experience(void);
+-extern void dump_experience(void);
++extern void dump_experience(const char*);
+ extern void free_experience(void);
+ /* friend.c */
+ extern void add_friendly_object(object *op);
+--- a/include/sproto.h
++++ b/include/sproto.h
+@@ -458,7 +458,7 @@
+ void quest_set_player_state(player *pl, sstring quest_code, int state);
+ int quest_was_completed(player *pl, sstring quest_code);
+ void command_quest(object *op, const char *params);
+-void dump_quests(void);
++void dump_quests(const char*);
+ void free_quest(void);
+ void free_quest_definitions(void);
+ void quest_send_initial_states(player *pl);
+--- a/server/init.c
++++ b/server/init.c
+@@ -33,7 +33,7 @@
+ #include "server.h"
+ #include "sproto.h"
+ 
+-static void help(void);
++static void help(const char*);
+ static void init_beforeplay(void);
+ static void init_startup(void);
+ static void init_signals(void);
+@@ -46,73 +46,73 @@
+  * Command line option: set logfile name.
+  * @param val new name.
+  */
+-static void set_logfile(char *val) {
++static void set_logfile(const char *val) {
+     settings.logfilename = val;
+ }
+ 
+ /** Command line option: show version. */
+-static void call_version(void) {
++static void call_version(const char*) {
```

Comment:
Unnamed parameters are a C23ism. While it's supported as a GNU extension in older C standards, older compilers don't support it (I think even GCC 11 doesn't).